src/gui/sdlmain.cpp

```diff
@@ -755,8 +753,40 @@ SDL_Rect GFX_GetSDLSurfaceSubwindowDims(Bit16u width, Bit16u height)
 	return rect;
 }
 
+static SDL_Window *setup_window_pp(SCREEN_TYPES screen_type, bool resizable)
+{
+	const SDL_Rect available = get_available_area(sdl.draw.width, sdl.draw.height);
+
+	if (!calc_pp_scale(available.w, available.h))
```

Comment:
Question - I'm guessing the `calc_pp_scale(..)` function populates the `sdl.pp_scale` global?

Another option would be to make `calc_pp_scale()` return an `SDL_Point` that we explicitly assign (just so it's more obvious): `sdl.pp_scale = calc_pp_scale(available.w, available.h)`.